source/Plugins/Language/ObjC/NSArray.cpp

```diff
@@ -35,17 +35,16 @@ using namespace lldb_private::formatters;
 
 namespace lldb_private {
 namespace formatters {
-std::map<ConstString, CXXFunctionSummaryFormat::Callback> &
+NSArray_Additionals::SummaryFormatters &
```

Comment:
This looks like it's code in `ObjC/`. Is this Swift specific? If so, maybe we should move it to a `Swift/` specific directory to reduce the amount of conflicts with upstream. Otherwise, if this might be useful for `NSArray` et similia also when running in `ObjC` language mode, this patch (or at least, part of it) should live upstream.

Reply:
These are hooks that Swift inserted so that it can hook into the Objective-C mode's formatter dispatch for Swift's implementations of Foundation collections. ObjC could use these to register its own formatters, too, but it doesn't really need it -- the if ladders it currently has work fine.

The new ObjC formatter (9b0178f172c284a507559d022374cead013fb152) and related NSSet formatter fixes (790e5f6cb1a954b0c778a54568932f6f198d11c6) are entirely Objective-C specific and should definitely live upstream!

Reply:
Ah, wait, I see the hooks are also in the upstream repo. You're right, the Objective-C parts of 66cbe5880522688825c93b91c51470c0aea2149a should also get merged there!

Reply:
I'd very much recommend to just submit the patch upstream (I'll be happy to review it!) and then cherry-pick here (at least, this is the flow we've been following for a while).